src/common/store/accounts.slice.ts

```diff
@@ -8,15 +8,22 @@ import type {
   SerializedDeterministicWallet,
   SerializedWallet,
   SerializedWalletWithAddress,
-  TAddress
+  TAddress,
+  WalletType
 } from '@types';
 
 export interface AccountsState {
   accounts: IAccount[];
-  accountsToAdd: SerializedWalletWithAddress[];
   addresses: GetAddressesResult[];
   isFetching: boolean;
   fetchError?: string;
+  add?: {
+    // Keystore files can be added as well, but for saving paper wallets we only support mnemonics
```

Comment:
Is there a reason to distinguish here? The secret of a keystore is the private key? Couldn't it still be typed as a `KEYSTORE`?

Reply:
Wouldn't it be clearer to keep the details of that in the PaperWallet implementation? I.e. supporting all types but inside PaperWallet `KEYSTORE` = `PRIVATE_KEY`?